### PR TITLE
[SEESAW] Port

### PR DESCRIPTION
**Pull request by networkfusion**
## Description

This PR updates #230 to resolve all merge conflicts and bring the project files up-to-date.
It also adds various fixes for issues found with style-cop.
I do not have a device to test, but the changes seem sensible.

## Motivation and Context

I would update the original PR, but it does not seem possible.
The original PR seems abandoned.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

## Summary by CodeRabbit

- **Chores**
	- Updated dependencies for `.NETnanoFramework` in the `devices/Seesaw` package to ensure compatibility and performance enhancements.

### CODE REVIEW

**Review by Ellerbach — COMMENTED** [review]
Looks great! You may have to adjust the fzz files and pictures there. Not sure what's left for you to work on with this port.

**Comment by networkfusion** [comment]
> Looks great! You may have to adjust the fzz files and pictures there. Not sure what's left for you to work on with this port.

Unfortunately I will have to leave the diagram updates and the testing to someone that has one.

**Comment by networkfusion** [comment]
@Ellerbach , can you recommend a way forward for this?  I am tempted to just remove the diagrams, so that it can be merged, and then if needed, improved in future...

**Comment by Ellerbach** [comment]
>  I am tempted to just remove the diagrams

fine for me!

**Comment by networkfusion** [comment]
Okay. AFIK, ready for a final review and hopeful merge.

**Review by Ellerbach — COMMENTED** [review]
Couple of comments bit looks good all up! Thanks!

File: devices/Seesaw/category.txt
[the remark this answers is not in the document]
Comment (networkfusion) [reply devices/Seesaw/category.txt:1]: image

File: devices/Seesaw/Seesaw.nuspec [hunk 1]
```diff
@@ -19,7 +19,15 @@
     <summary>Iot.Device.Seesaw assembly for .NET nanoFramework C# projects</summary>
     <tags>nanoFramework C# csharp netmf netnf Iot.Device.Seesaw</tags>
     <dependencies>
-      <dependency id="nanoFramework.CoreLibrary" version="1.10.4-preview.11" />
+      <dependency id="nanoFramework.CoreLibrary" version="1.15.5" />
+      <dependency id="nanoFramework.Runtime.Events" version="1.11.18" />
```
[the remark this answers is not in the document]
Comment (networkfusion) [reply devices/Seesaw/Seesaw.nuspec]: Done.

File: devices/Seesaw/SeesawModules.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            Keypad = 0x10,
+
+            /// <summary>Encoder module.</summary>
+            Encoder = 0x11
+        }
+
+        /// <summary>
+        /// Seesaw function.
+        /// </summary>
+        protected enum SeesawFunction : byte
```
[the remark this answers is not in the document]
Comment (networkfusion) [reply devices/Seesaw/SeesawModules.cs]: Done.

File: devices/Seesaw/samples/Seesaw.Sample.BlinkingLights/packages.config [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="nanoFramework.CoreLibrary" version="1.15.5" targetFramework="netnano1.0" />
+  <package id="nanoFramework.Runtime.Events" version="1.11.18" targetFramework="netnano1.0" />
+  <package id="nanoFramework.System.Device.Gpio" version="1.1.41" targetFramework="netnano1.0" />
+  <package id="nanoFramework.System.Device.I2c" version="1.1.16" targetFramework="netnano1.0" />
+  <package id="nanoFramework.System.Math" version="1.5.43" targetFramework="netnano1.0" />
```
[the remark this answers is not in the document]
Comment (networkfusion) [reply devices/Seesaw/samples/Seesaw.Sample.BlinkingLights/packages.config:7]: Why? these are in addition to the seesaw nuget ref...

File: devices/Seesaw/README.md [hunk 4]
```diff
@@ -19,36 +19,11 @@ This binding was developed using the Adafruit Seesaw breakout board which uses t
 
 ## Usage
 
-These samples connect to a Raspberry Pi via the first I2C interface. Issues were noticed when using the out-of-the-box stiings and so the I2C bus has been slowed down by adding the following to the /boot/Config.txt file
-
- `dtparam=i2c1_baudrate=50000`
+These samples connect to an ESP32 via the first I2C interface.
```
[the remark this answers is not in the document]
Comment (networkfusion) [reply devices/Seesaw/README.md:23]: Done.

File: devices/Seesaw/samples/Seesaw.Sample.BlinkingLights/packages.config (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ellerbach) [reply devices/Seesaw/samples/Seesaw.Sample.BlinkingLights/packages.config:7]: some are dependents of others, so they are pulled in and the version does not need to be maintained in bumps. In short: it's a bit easier to manage. But it's ok :-) I think only runtime events is in the gpio one.

**Review by Ellerbach — APPROVED** [review]
I think we're good :-)

**Pull request merged**